atc/db/resource_config_factory.go

```diff
@@ -109,22 +108,17 @@ func (f *resourceConfigFactory) FindOrCreateResourceConfig(
 func constructResourceConfigDescriptor(
 	resourceTypeName string,
 	source atc.Source,
-	resourceTypes creds.VersionedResourceTypes,
+	resourceTypes atc.VersionedResourceTypes,
 ) (ResourceConfigDescriptor, error) {
 	resourceConfigDescriptor := ResourceConfigDescriptor{
 		Source: source,
 	}
 
 	customType, found := resourceTypes.Lookup(resourceTypeName)
 	if found {
-		source, err := customType.Source.Evaluate()
```

Comment:
One concern I have here is that whereas before we would only evaluate the resource type that ends up being used, now we evaluate *all* of them.

Probably not a huge deal. Most resource types don't need credentials. :man_shrugging:

Reply:
I think the mitigation here is to just enable caching. That resource type is going to be actually evaluated and used *somewhere*, so it'll have been fetched anyway, and if you're using credential caching it'll only hit the credential manager once regardless of how many places call `.Evaluate` within that minute.